packages/kbn-esql-validation-autocomplete/scripts/generate_function_definitions.js

```diff
@@ -0,0 +1,338 @@
+};
+
+const excludedFunctions = new Set(['bucket', 'case']);
+
+const extraFunctions = [
+  {
+    type: 'eval',
+    name: 'case',
+    description:
+      'Accepts pairs of conditions and values. The function returns the value that belongs to the first condition that evaluates to `true`. If the number of arguments is odd, the last argument is the default value which is returned when no condition matches.',
+    ...evalSupportedCommandsAndOptions,
+    signatures: [
+      {
+        params: [
+          { name: 'condition', type: 'boolean' },
+          { name: 'value', type: 'any' },
+        ],
+        minParams: 2,
+        returnType: 'any',
+        examples: [
+          `from index | eval type = case(languages <= 1, "monolingual", languages <= 2, "bilingual", "polyglot")`,
+        ],
+      },
+    ],
+  },
+];
+
+const elasticsearchToKibanaType = (elasticsearchType) => {
+  if (
+    [
+      'double',
+      'unsigned_long',
+      'long',
+      'integer',
+      'counter_integer',
+      'counter_long',
+      'counter_double',
+    ].includes(elasticsearchType)
+  ) {
+    return 'number';
+  }
+
+  if (['text', 'keyword'].includes(elasticsearchType)) {
+    return 'string';
+  }
+
+  if (['datetime', 'time_duration'].includes(elasticsearchType)) {
+    return 'date';
+  }
+
+  if (elasticsearchType === 'date_period') {
+    return 'time_literal'; // TODO - consider aligning with Elasticsearch
+  }
+
+  return elasticsearchType;
+};
+
+const validateLogFunctions = `(fnDef: ESQLFunction) => {
+  const messages = [];
+  // do not really care here about the base and field
+  // just need to check both values are not negative
+  for (const arg of fnDef.args) {
+    if (isLiteralItem(arg) && arg.value < 0) {
+      messages.push({
+        type: 'warning' as const,
+        code: 'logOfNegativeValue',
+        text: i18n.translate(
+          'kbn-esql-validation-autocomplete.esql.divide.warning.logOfNegativeValue',
+          {
+            defaultMessage: 'Log of a negative number results in null: {value}',
+            values: {
+              value: arg.value,
+            },
+          }
+        ),
+        location: arg.location,
+      });
+    }
+  }
+  return messages;
+}`;
+
+const dateDiffSuggestions = [
+  'year',
+  'quarter',
+  'month',
+  'week',
+  'day',
+  'hour',
+  'minute',
+  'second',
+  'millisecond',
+  'microsecond',
+  'nanosecond',
+];
+
+const dateDiffOptions = [
+  'year',
+  'years',
+  'yy',
+  'yyyy',
+  'quarter',
+  'quarters',
+  'qq',
+  'q',
+  'month',
+  'months',
+  'mm',
+  'm',
+  'dayofyear',
+  'dy',
+  'y',
+  'day',
+  'days',
+  'dd',
+  'd',
+  'week',
+  'weeks',
+  'wk',
+  'ww',
+  'weekday',
+  'weekdays',
+  'dw',
+  'hour',
+  'hours',
+  'hh',
+  'minute',
+  'minutes',
+  'mi',
+  'n',
+  'second',
+  'seconds',
+  'ss',
+  's',
+  'millisecond',
+  'milliseconds',
+  'ms',
+  'microsecond',
+  'microseconds',
+  'mcs',
+  'nanosecond',
+  'nanoseconds',
+  'ns',
+];
```

Comment:
Could consider storing some of this info in files imported into the generated file. I like it here for now...